zrml/asset-router/src/pallet_impl/multicurrency.rs

```diff
@@ -46,14 +46,21 @@ impl<T: Config> MultiCurrency<T::AccountId> for Pallet<T> {
     }
 
     fn free_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
-        if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
-            <T::Currencies as MultiCurrency<T::AccountId>>::free_balance(currency, who)
-        } else if let Ok(asset) = T::MarketAssetType::try_from(currency_id) {
-            T::MarketAssets::reducible_balance(asset, who, false)
+        if let Ok(asset) = T::MarketAssetType::try_from(currency_id) {
+            // Route "pre new asset system" market assets to `CurrencyType`
+            if T::MarketAssets::asset_exists(asset) {
+                T::MarketAssets::reducible_balance(asset, who, false)
+            } else if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
+                T::Currencies::free_balance(currency, who)
```

Comment:
AFAIU, this is reached if `free_balance` is called with a market asset (meaning anything that successfully converts into a `T::MarketAssetType` object) which isn't registered. We seem to be missing this because we're using `ForeignAsset` for testing.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1256/commits/9df31a5733ef97dd3b1f53be6c92ab9a22adc636